app/models/host/discovered.rb

```diff
@@ -1,6 +1,9 @@
 class Host::Discovered < ::Host::Base
   include ScopedSearchExtensions
   include Foreman::Renderer
+  include BelongsToProxies
+
+  self.registered_smart_proxies = {}
```

Comment:
I don't like this, why I don't see it in core? Isn't better to initialize the class member in the `BelongsToProxies` concern?